core/GlobalState.cc

```diff
@@ -113,6 +113,79 @@ unique_ptr<GlobalState> GlobalState::makeEmptyGlobalStateForHashing(spdlog::logg
     return rv;
 }
 
+struct MethodBuilder {
+    GlobalState &gs;
+    MethodRef method;
+
+    MethodBuilder(GlobalState &gs, MethodRef m) : gs(gs), method(m) {}
+
+    MethodBuilder &defaultArg(NameRef name) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.flags.isDefault = true;
+        return *this;
+    }
+
+    MethodBuilder &blockArg() {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, Names::blkArg());
+        arg.flags.isBlock = true;
+        return *this;
+    }
+
+    MethodBuilder &typedArg(NameRef name, TypePtr &&type) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.type = std::move(type);
+        return *this;
+    }
+
+    MethodBuilder &arg(NameRef name) {
+        gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        return *this;
+    }
+
+    MethodBuilder &untypedArg(NameRef name) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.type = Types::untyped(gs, method);
+        return *this;
+    }
+
+    MethodBuilder &repeatedArg(NameRef name) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.flags.isRepeated = true;
+        return *this;
+    }
+
+    MethodBuilder &repeatedTypedArg(NameRef name, TypePtr &&type) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.flags.isRepeated = true;
+        arg.type = std::move(type);
+        return *this;
+    }
+
+    MethodBuilder &repeatedUntypedArg(NameRef name) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.flags.isRepeated = true;
+        arg.type = Types::untyped(gs, method);
+        return *this;
+    }
+
+    MethodBuilder &kwsplatArg(NameRef name) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.flags.isKeyword = true;
+        arg.flags.isRepeated = true;
+        arg.type = Types::untyped(gs, method);
+        return *this;
+    }
+
+    // Auto-conversion when we're done.
+    operator MethodRef() {
+        return method;
+    }
+};
+
+MethodBuilder def(GlobalState &gs, ClassOrModuleRef klass, NameRef name) {
+    return MethodBuilder{gs, gs.enterMethodSymbol(Loc::none(), klass, name)};
+}
```

Comment:
What do you think about giving this method a slightly longer / more descriptive name?

Reply:
WFM.  I was going for Ruby-ish, but I changed this to `enterMethod`.